File: algorithms/M/Merge Similar Items/Merge Similar Items.js
/**
 * @param {number[][]} items1
 * @param {number[][]} items2
 * @return {number[][]}
 */
var mergeSimilarItems = function(items1, items2) {
    let map = new Map();
    items1.reduce((map,curr)=>map.set(curr[0],map.get(curr[0])+curr[1]||curr[1]),map);
    items2.reduce((map,curr)=>map.set(curr[0],map.get(curr[0])+curr[1]||curr[1]),map);
    
    return [...map].sort((a,b)=>{
        return a[0] - b[0];
    })
};

File: algorithms/M/Merge Similar Items/Merge Similar Items.cpp
class Solution {
public:
    vector<vector<int>> mergeSimilarItems(vector<vector<int>>& items1, vector<vector<int>>& items2) {
        map<int, int> m;
        
        for(auto v : items1) m[v[0]] += v[1];
        for(auto v : items2) m[v[0]] += v[1];
        
        vector<vector<int>> ans;
        for(auto i : m) ans.push_back({i.first, i.second});
        
        return ans;
    }
};
